test/test_datasets_samplers.py

```diff
@@ -13,34 +13,33 @@
 from torchvision.datasets.video_utils import VideoClips, unfold
 from torchvision import get_video_backend
 
-from common_utils import get_tmp_dir, assert_equal
+from common_utils import assert_equal
 
 
 @contextlib.contextmanager
-def get_list_of_videos(num_videos=5, sizes=None, fps=None):
-    with get_tmp_dir() as tmp_dir:
-        names = []
-        for i in range(num_videos):
-            if sizes is None:
-                size = 5 * (i + 1)
-            else:
-                size = sizes[i]
-            if fps is None:
-                f = 5
-            else:
-                f = fps[i]
-            data = torch.randint(0, 256, (size, 300, 400, 3), dtype=torch.uint8)
-            name = os.path.join(tmp_dir, "{}.mp4".format(i))
-            names.append(name)
-            io.write_video(name, data, fps=f)
+def get_list_of_videos(tmpdir, num_videos=5, sizes=None, fps=None):
+    names = []
+    for i in range(num_videos):
+        if sizes is None:
+            size = 5 * (i + 1)
+        else:
+            size = sizes[i]
+        if fps is None:
+            f = 5
+        else:
+            f = fps[i]
+        data = torch.randint(0, 256, (size, 300, 400, 3), dtype=torch.uint8)
+        name = os.path.join(tmpdir, "{}.mp4".format(i))
+        names.append(name)
+        io.write_video(name, data, fps=f)
 
-        yield names
+    yield names
 
 
 @pytest.mark.skipif(not io.video._av_available(), reason="this test requires av")
 class TestDatasetsSamplers:
-    def test_random_clip_sampler(self):
-        with get_list_of_videos(num_videos=3, sizes=[25, 25, 25]) as video_list:
+    def test_random_clip_sampler(self, tmpdir):
+        with get_list_of_videos(tmpdir, num_videos=3, sizes=[25, 25, 25]) as video_list:
```

Comment:
I think that `get_list_of_videos` was written as a context manager because it was relying on `get_tmp_dir()` which was itself a context manager.

But now that we're getting rid of `get_tmp_dir()`, I believe we should be able to convert `get_list_of_videos()` into a simple function: we can remove the `@contextlib.contextmanager` decorator and change `yield` into a `return`. And here, we could just do 

```py
video_list = get_list_of_videos(tmpdir, num_videos=3, sizes=[25, 25, 25])
```

and indent everything to the left

Reply:
Thanks, edited. :)